Fix travis CI setup

rbertran opened the pull request:
Travis CI has more strict job limits than the Travis enterprise version. That configuration miss-match lead to time-out CI failures in the public Travis CI runs that were undetected in IBM internal ones.  This commit fixes the issues by reducing the work done on the steps that were timing-out. In particular, now the risc-v custom toolchain is built without GDB and QEMU support.   

Signed-off-by: Ramon Bertran Monfort <rbertra@us.ibm.com>

seldridge commented:
If this isn't the hard job time limit, but the "no message received in X minutes", you may be able to use `travis_wait` to get around it: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received.

rbertran commented:
@seldridge thanks for the pointer but the issue was related to the 50 minutes wall-time timeout per job. I split the CI in various steps and implemented caching to avoid re-compiling the tool-chain every time. Now it seems that it's working. 

Now, I'm worried about caching. I'm not sure if the cache expires after some time or I've to manually delete it to force a 'refresh', I need to check that... but probably it expires after some time without being updated (it is what makes more sense).
